experts-api/lib/elements-client.js

```diff
@@ -39,17 +39,26 @@ export default class ElementsClient {
         delete ElementsClient.impersonators[userId];
       }
     }
-    let user=new Impersonator(userId,args);
-    await user.login();
-    await user.impersonate();
-    ElementsClient.impersonators[userId]=user;
-    return user
+    try {
+      let user=new Impersonator(userId,args);
+      await user.login();
+      await user.impersonate();
+      ElementsClient.impersonators[userId]=user;
+      return user
+    } catch (e) {
+      if (e.status !== 504) {
+        e.message = `Impersonation failed: ${e.message}`;
```

Comment:
# Example 4 #
And finally this example.  The impersonation function calls about 4 different fetch requests.  These are also handled by the `fetchWithTimeout` function, but we are catching any error from there, and just noting that we can't `impersonate` an expert.